File: src/main/java/io/github/shaksternano/borgar/media/io/MediaWriterFactory.java
package io.github.shaksternano.borgar.media.io;

import io.github.shaksternano.borgar.media.io.writer.MediaWriter;

import java.io.File;
import java.io.IOException;

@FunctionalInterface
public interface MediaWriterFactory {

    MediaWriter createWriter(
        File output,
        String outputFormat,
        int audioChannels,
        int audioSampleRate,
        int audioBitrate,
        long maxFileSize,
        long maxDuration
    ) throws IOException;
}

File: core/src/main/kotlin/io/github/shaksternano/borgar/core/media/MediaWriterFactory.kt
package io.github.shaksternano.borgar.core.media

import io.github.shaksternano.borgar.core.media.writer.MediaWriter
import java.nio.file.Path
import kotlin.time.Duration

interface MediaWriterFactory {

    val supportedFormats: Set<String>

    suspend fun create(
        output: Path,
        outputFormat: String,
        loopCount: Int,
        audioChannels: Int,
        audioSampleRate: Int,
        audioBitrate: Int,
        maxFileSize: Long,
        maxDuration: Duration,
    ): MediaWriter
}
